torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -72,9 +72,13 @@ def parallelize_llama(
         apply_compile(model)
 
     if parallel_dims.dp_enabled:
-        if parallel_dims.dp_type == "fsdp":
-            dp_mesh = world_mesh["dp"] if world_mesh.ndim > 1 else world_mesh
-            assert dp_mesh.mesh_dim_names == ("dp",), dp_mesh.mesh_dim_names
+        if parallel_dims.dp_type in ("fsdp", "hsdp"):
```

Comment:
i just wonder, do we need to have concepts of "fsdp, hsdp" in the top level at all?  

What if we just let users specify dp_shard_degree and dp_replicate_degree args?

Side question: do we support pure DDP?  i guess its by setting 'hsdp' and replicate == dp_degree?

Reply:
We do support DDP.  There are two different ways to support this, one is the `dp_type` and another one is exposing `dp_shard_degree` and `dp_replicate_degree` as you mentioned. I proposed both earlier and the discussion led to conclustion to use `dp_type` instead of `dp_shard_degree` and `dp_replicate_degree`.  cc., @tianyu-l, @wanchaol

Reply:
IIRC @wanchaol suggested we use a single field of int/tuple to denote both types of degrees
https://github.com/pytorch/torchtitan/pull/319/files#r1596294918

Is this even possible for config_manager?

Reply:
oh, right, that's another good way to do so. But we still have to expose HSDP even if we don't expose dp_replicate to users, which seems to be the point from @wconstab's comment

Reply:
hot take: we just expose `dp_replicate` and `dp_shard` to  users.  We document in the readme and the argparse help, something like:  "DDP=(dp_replicate, 1), FSDP = (1, dp_shard), HSDP = (dp_replicate, dp_shard)"

we don't need a "type" argument.

cc @tianyu-l @fegin wdyt